anxcloud/data_source_core_location.go

```diff
@@ -0,0 +1,80 @@
+package anxcloud
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	corev1 "go.anx.io/go-anxcloud/pkg/apis/core/v1"
+)
+
+func dataSourceCoreLocation() *schema.Resource {
+	return &schema.Resource{
+		Description: "Retrieves Location identified by it's `code` as selectable in the Engine.",
+		ReadContext: dataSourceCoreLocationRead,
+		Schema:      schemaLocation(),
+	}
+}
+
+func dataSourceCoreLocationRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	a := m.(providerContext).api
+
+	code, exists := d.GetOk("code")
+	if !exists {
+		return diag.Errorf("location data-source requires code argument")
+	}
+
+	listCTX, cancel := context.WithCancel(ctx)
```

Comment:
can't explain it, but `listCTX` looks odd to me - maybe just replace `ctx` with your new context?

Up to you, not preventing a merge :)